### PR TITLE
Add the HTTP/2 ServerSelector

### PR DESCRIPTION
**Pull request by bryce-anderson**
The final piece needed for making HTTP/2 server sessions

### CODE REVIEW

**Comment by bryce-anderson** [comment]
Note that this depends upon #153.

**Comment by bryce-anderson** [comment]
Also, this will likely be the last commit before I go about merging the `merge-new-client-http2` branch into `master`. :)

File: examples/src/main/scala/org/http4s/blaze/examples/http2/Http2ServerExample.scala [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  }
+
+  private val factory = NIO1SocketServerGroup.fixedGroup(workerThreads = channel.DefaultPoolSize)
+
+  def run(): ServerChannel = factory.bind(new InetSocketAddress(port), f).getOrElse(sys.error("Failed to start server."))
+}
+
+object Http2ServerExample {
+  def main(args: Array[String]): Unit = {
+    println("Hello world!")
```
Comment (ChristopherDavenport) [inline examples/src/main/scala/org/http4s/blaze/examples/http2/Http2ServerExample.scala]: Whats this `println` about? Just to signify execution right before the server should start?

Comment (bryce-anderson) [reply examples/src/main/scala/org/http4s/blaze/examples/http2/Http2ServerExample.scala]: Yeah, and it should be something else more meaningful. Thanks for pointing it out.

**Comment by bryce-anderson** [comment]
@ChristopherDavenport, don't flatter me. 😉  There shouldn't be anything particularly tricky here, so if it was clear then that is a great sign.

**Pull request merged**